File: testdata/python/secure.py
"""Properly configured TLS - should not trigger critical/high findings."""

import ssl

# Certificate verification must always be enabled.
# Use PROTOCOL_TLS_CLIENT instead of legacy protocol constants.

def create_secure_context():
    ctx = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
    ctx.verify_mode = ssl.CERT_REQUIRED
    ctx.check_hostname = True
    ctx.load_default_certs()
    return ctx

def log_status():
    print("TLS verification is active")
    print("Using secure cipher configuration")

File: testdata/java/Secure.java
package secure;

import javax.net.ssl.SSLContext;
import javax.net.ssl.HttpsURLConnection;

// Properly configured TLS - should not trigger critical/high findings.
// Always use versioned SSLContext and default hostname verification.

public class Secure {
    public static SSLContext createSecureContext() throws Exception {
        SSLContext ctx = SSLContext.getInstance("TLSv1.3");
        ctx.init(null, null, null);
        return ctx;
    }

    public static void logStatus() {
        System.out.println("Certificate verification is enabled");
        System.out.println("Hostname verification is active");
    }
}
